internal/controller/apiextensions/composite/reconciler.go

```diff
@@ -158,6 +158,11 @@ type CompositionResult struct {
 	ConnectionDetails managed.ConnectionDetails
 	Events            []TargetedEvent
 	Conditions        []TargetedCondition
+
+	// ConditionsOverride allows the composition process to force
+	// a specific condition status, unlike Conditions which does not allow
+	// setting system conditions.
+	ConditionsOverride []xpv1.Condition
```

Comment:
I wonder if we could use a special `TargetdCondition` variant. i.e. Add a `System` field and just use that internally to surface these conditions.

Reply:
I tried this but it does work as intended since the custom conditions update (which uses the `Conditions` array) is performed in `handleCommonCompositionResult()` but the system conditions are set in `updateXRConditions()`.

Reply:
I changed to a new type `CompositeResource` that is analog to `ComposedResource` and contains a `Ready *bool` property.